script/tool/lib/src/common/core.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:colorize/colorize.dart';
+import 'package:file/file.dart';
+import 'package:yaml/yaml.dart';
+
+/// The signature for a print handler for commands that allow overriding the
+/// print destination.
+typedef Print = void Function(Object? object);
+
+/// Key for windows platform.
+const String kPlatformFlagWindows = 'windows';
+
+/// Key for macos platform.
+const String kPlatformFlagMacos = 'macos';
+
+/// Key for linux platform.
+const String kPlatformFlagLinux = 'linux';
+
+/// Key for IPA (iOS) platform.
+const String kPlatformFlagIos = 'ios';
+
+/// Key for APK (Android) platform.
+const String kPlatformFlagAndroid = 'android';
+
+/// Key for Web platform.
+const String kPlatformFlagWeb = 'web';
+
+/// Key for enable experiment.
+const String kEnableExperiment = 'enable-experiment';
```

Comment:
nits: maybe we can refactor all these to a file called constants.dart?

Reply:
I wanted to hold off on making a "constants" file until tool refactoring is further along; I think in most cases there will be a more specific home that makes sense. (Even these are a bit iffy in that they are used multiple ways which might be better managed via usage-specific enums, so I'm not sure they will survive as-is.)